app/models/piece.rb

```diff
@@ -12,31 +12,33 @@ def move!(x,y)
     if valid_move?(x,y)
       self.x_position = x
       self.y_position = y
+      capture!(x,y)
       return true
     else
       return false
     end
   end
 
-  #Inital Capture Logic:
-  #Need game model to test id of game
-
-  #Set initial captured status to false
-  def default_captured_status
-    self.captured = false if self.captured = nil
+  # First check if coordinates is taken
+  # return false if coordinates belongs to friendly piece
+  # return true if target piece's attributes changes
+  def capture!(x,y)
+    if position_taken?(x,y)
+      return false if friendly_piece?(x,y)
+      return true if target_piece(x,y).update!(x_position: nil, y_position: nil, captured: true)
+    end
   end
 
-  #Find the target piece in the database base on the x,y coordinates
-  def capturable?(x,y)
-    Piece.where(:x_position => x, :y_position => y, :captured => false).where.not(:color => color).present?
+  # Return true if target is a friendly piece
+  # otherwise return false
+  def friendly_piece?(x, y)
+    return true if color == target_piece(x,y).color
+    false
   end
 
-  #When the target is captured, change it's attributes and remove it from the board
-  def capture!(x,y)
-    opponent_piece = Piece.where(:x_position => x, :y_position => y)
-    if capturable?(x,y)
-      opponent_piece[0].update_attributes(:x_position => nil, :y_position => nil, :captured => true)
-    end
+  # Find the target piece base on game id, and x,y coordinates
+  def target_piece(x, y)
+    Piece.where(game_id: game_id, x_position: x, y_position: y).take
```

Comment:
Use find_by instead of where.  It will return the piece instead of an activerecord association so you don't need to use take.